src/renderer/views/Channel/Channel.js

```diff
@@ -2009,6 +2036,37 @@ export default defineComponent({
       })
     },
 
+    async saveStateInRouter(query) {
+      this.skipRouteChangeWatcherOnce = true
+      if (query === '') {
+        await this.$router.replace({ path: `/channel/${this.id}` }).catch(failure => {
+          if (isNavigationFailure(failure, NavigationFailureType.duplicated)) {
+            return
+          }
+
+          throw failure
+        })
```

Comment:
To make the code more readable please pick one of the Promise syntaxes either async-await or chaining, instead of combining both of them. When using async-await, you can just use a normal try-catch.

This suggestion applies to all files that have this construct introduced by this pull request.